File: src/main/java/memo/model/PaymentMethod.java
package memo.model;


import java.util.Arrays;

public enum PaymentMethod {
    Bar("Bar"),
    Lastschrift("Lastschrift"),
    Überweisung("Überweisung"),
    Paypal("Paypal");

    String textValue;

    PaymentMethod(String textValue) {
        this.textValue = textValue;
    }

    public String getTextValue() {
        return textValue;
    }

    public static PaymentMethod fromTextValue(String textValue) {
        return Arrays.stream(PaymentMethod.values())
                .filter(it -> it.getTextValue().equalsIgnoreCase(textValue))
                .findFirst()
                .orElse(null);
    }
}

File: src/main/web/src/app/shop/checkout/payment/payment-method.ts
export enum PaymentMethod {
	CASH = "Bar",
	// PAYPAL = <any> "Paypal",
	TRANSFER = "Überweisung",
	DEBIT = "Lastschrift"
}

const paymentMethodMap = {
	0: PaymentMethod.CASH,
	1: PaymentMethod.DEBIT,
	2: PaymentMethod.TRANSFER,
	// 3: PaymentMethod.PAYPAL
};

export function toPaymentMethod(index: number) {
	return paymentMethodMap[index];
}

export function paymentMethodList() {
	return Object.keys(paymentMethodMap).map(key => paymentMethodMap[key]);
}
